kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/symbol/kotlin/KSPropertyAccessorImpl.kt

```diff
@@ -50,23 +51,33 @@ abstract class KSPropertyAccessorImpl(
         get() = TODO("Not yet implemented")
 }
 
-class KSPropertySetterImpl(
+class KSPropertySetterImpl private constructor(
     owner: KSPropertyDeclaration,
     private val setter: KtPropertySetterSymbol
 ) : KSPropertySetter, KSPropertyAccessorImpl(setter, owner) {
+    companion object : KSObjectCache<Pair<KSPropertyDeclaration, KtPropertySetterSymbol>, KSPropertySetterImpl>() {
+        fun getCached(owner: KSPropertyDeclaration, setter: KtPropertySetterSymbol) = cache
+            .getOrPut(Pair(owner, setter)) { KSPropertySetterImpl(owner, setter) }
+    }
+
     override val parameter: KSValueParameter by lazy {
-        KSValueParameterImpl(setter.parameter)
+        KSValueParameterImpl.getCached(setter.parameter)
     }
 
     override fun <D, R> accept(visitor: KSVisitor<D, R>, data: D): R {
         return visitor.visitPropertySetter(this, data)
     }
 }
 
-class KSPropertyGetterImpl(
+class KSPropertyGetterImpl private constructor(
     owner: KSPropertyDeclaration,
     getter: KtPropertyGetterSymbol
```

Comment:
Just curious; `KSPropertySetterImpl.setter` above is `private`, but why this one is not?

Reply:
They should both not be `val`